tests/plugins/subcommands/doctor/test_health_checks.py

```diff
@@ -83,6 +85,22 @@ def env_altered_files(env_ok: tuple[Path, str, str, str]) -> tuple[Path, str, st
     return env_ok
 
 
+def test_listed_on_envs_txt_file(
+    tmp_path: Path, mocker: MockerFixture, env_ok: tuple[Path, str, str, str]
+):
+    """Test that runs for the case when the env is listed on the environments.txt file"""
+    prefix, _, _, _ = env_ok
+    tmp_envs_txt_file = tmp_path / "envs.txt"
+    with open(tmp_envs_txt_file, "w") as f:
+        f.write(f"{prefix}")
+
+    mocker.patch(
+        "conda.core.envs_manager.get_user_environments_txt_file",
+        return_value=tmp_envs_txt_file,
+    )
```

Comment:
This actually patches the "wrong" function.

When we import a function as done above:

```python
from conda.core.envs_manager import get_user_environments_txt_file
```

We end up making a unique _pointer_ to that specific instance of the `get_user_environments_txt_file` function.

This means that when we mock the actual `get_user_environments_txt_file` function only the function defined at `conda.core.envs_manager` is mocked and not the one at `conda.plugins.subcommands.doctor.health_checks`.

So there are two solutions to this, (1) mock the `get_user_environments_txt_file` in `conda.plugins.subcommands.doctor.health_checks`:

```suggestion
    mocker.patch(
        "conda.plugins.subcommands.doctor.health_checks.get_user_environments_txt_file",
        return_value=tmp_envs_txt_file,
    )
```

Or (2) change the import in `conda.plugins.subcommands.doctor.health_checks` to be:

```python
from conda.core import envs_manager
...
envs_txt_file = envs_manager.get_user_environments_txt_file()
```